### PR TITLE
releng: Grant Release Manager Admins artifact admin privileges

### PR DESCRIPTION
**Pull request by justaugustus**
The group of contributors defined in `k8s-infra-release-admins` (@kubernetes/sig-release-admins -- @justaugustus @tpepper @saschagrunert @alejandrox1) has extended admin privileges (including KMS access) over Release Engineering GCP projects, defined here: https://github.com/kubernetes/sig-release/blob/master/release-engineering/gcp.md

As part of extending the scope of personnel who can handle artifact promotion, auditing, and overall job/infra maintenance (ref: https://github.com/kubernetes/k8s.io/issues/157), this PR does the following:

- Move `k8s-staging-cip-test` staging project under SIG Release review
- Grants @kubernetes/release-managers access to the `k8s-staging-cip-test` staging project
- Grants @kubernetes/sig-release-admins artifact admin privileges by way of access to the following IAM groups:
  - k8s-infra-artifact-admins

/hold for any discussion
/assign @thockin @spiffxp @listx @dims 
cc: @kubernetes/k8s-infra-team @kubernetes/release-engineering

### CODE REVIEW

**Review by hasheddan — APPROVED** [review]
/lgtm

File: groups/groups.yaml [hunk 1]
```diff
@@ -846,6 +849,7 @@ groups:
     settings:
       ReconcileMembers: "true"
     members:
+      - k8s-infra-release-editors@kubernetes.io
```
Comment (spiffxp) [inline groups/groups.yaml:852]: Fine with this; maybe `k8s-infra-staging-cip-test` should be moved to sig-release/groups.yaml?

File: groups/groups.yaml [hunk 2]
```diff
@@ -277,6 +279,7 @@ groups:
     settings:
       ReconcileMembers: "true"
     members:
+      - k8s-infra-release-admins@kubernetes.io
```
Comment (spiffxp) [inline groups/groups.yaml:282]: I would like to understand more about why membership in this group
- what does it allow?
- why is that needed?

File: groups/groups.yaml [hunk 3]
```diff
@@ -263,6 +264,7 @@ groups:
...
```
Comment (spiffxp) [inline groups/groups.yaml:267]: Mostly fine with this (SIG Release ownership + appropriately paranoid admins), though I would like to refresh my memory on what this allows

**Review by spiffxp — COMMENTED** [review]
Since it's taking a while for me to page in appropriate context, here are my thoughts off the cuff

File: groups/groups.yaml (hunk 1 above)
Comment (justaugustus) [reply groups/groups.yaml]: Done!

File: groups/groups.yaml (hunk 2 above)
Comment (justaugustus) [reply groups/groups.yaml]: Grants [`roles/containeranalysis.occurrences.viewer`](https://cloud.google.com/container-registry/docs/ca-access-control#iam_roles), which I don't think is strictly required here. I've dropped it from the PR.

ref: https://github.com/kubernetes/k8s.io/blob/ed671b680ce43ecc024d56df224bb85a8ce1a36e/infra/gcp/ensure-prod-storage.sh#L238-L245

File: groups/groups.yaml (hunk 3 above)
Comment (justaugustus) [reply groups/groups.yaml:266]: Artifact admins seem to get...

GCS Admin access:
https://github.com/kubernetes/k8s.io/blob/ed671b680ce43ecc024d56df224bb85a8ce1a36e/infra/gcp/lib.sh#L26-L28

https://github.com/kubernetes/k8s.io/blob/ed671b680ce43ecc024d56df224bb85a8ce1a36e/infra/gcp/lib.sh#L294-L308

https://github.com/kubernetes/k8s.io/blob/ed671b680ce43ecc024d56df224bb85a8ce1a36e/infra/gcp/lib.sh#L310-L323

https://github.com/kubernetes/k8s.io/blob/ed671b680ce43ecc024d56df224bb85a8ce1a36e/infra/gcp/ensure-prod-storage.sh#L117-L121

https://github.com/kubernetes/k8s.io/blob/ed671b680ce43ecc024d56df224bb85a8ce1a36e/infra/gcp/ensure-prod-storage.sh#L155-L156

https://github.com/kubernetes/k8s.io/blob/ed671b680ce43ecc024d56df224bb85a8ce1a36e/infra/gcp/ensure-prod-storage.sh#L201-L202

https://github.com/kubernetes/k8s.io/blob/ed671b680ce43ecc024d56df224bb85a8ce1a36e/infra/gcp/ensure-prod-storage.sh#L207-L208

https://github.com/kubernetes/k8s.io/blob/ed671b680ce43ecc024d56df224bb85a8ce1a36e/infra/gcp/ensure-prod-storage.sh#L211-L220

https://github.com/kubernetes/k8s.io/blob/ed671b680ce43ecc024d56df224bb85a8ce1a36e/infra/gcp/ensure-prod-storage.sh#L390-L392

---

Cloud Run access to deploy the auditor:
https://github.com/kubernetes/k8s.io/blob/ed671b680ce43ecc024d56df224bb85a8ce1a36e/infra/gcp/ensure-prod-storage.sh#L315-L324

File: groups/groups.yaml [hunk 4]
```diff
@@ -245,6 +245,7 @@ groups:
       WhoCanPostMessage: "ANYONE_CAN_POST"
       ReconcileMembers: "true"
     members:
+      - k8s-infra-release-admins@kubernetes.io
```
[the remark this answers is not in the document]
Comment (justaugustus) [reply groups/groups.yaml]: This is related to https://github.com/kubernetes/k8s.io/issues/157#issuecomment-718043478:

> Ensure that the community (someone who isn't @listx or @thockin) is able to respond to an alert, possibly proven via some kind of DiRT (Disaster Recovery Training) scenario

So I believe these would be alerts coming directly from GCP, not Testgrid (since the auditor runs from Cloud Run).

Staring at https://github.com/kubernetes/k8s.io/blob/master/k8s.gcr.io/Vanity-Domain-Flip.md#alerts:

> The Stackdriver Error Reporting leg of the auditing process is responsible for sending alerts to humans about the rejected GCR change.
>
> Currently, an email is sent to the project owner(s) of the k8s-artifacts-prod GCP project.

Implies that we might not have an appropriately-scoped group today to send auditor alerts to.

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm
/hold cancel

**Comment by justaugustus** [comment]
Thanks everyone for the thoughtful review! :)

File: groups/groups.yaml (hunk 3 above)
Comment (spiffxp) [reply groups/groups.yaml:266]: Thanks, crawling through this was fun.  

Granting iam.serviceAccountUsers for the auditor service account means this group can also pick up whatever that service account can do
https://github.com/kubernetes/k8s.io/blob/07a3a07626d8733550866abe799cf58241045a18/infra/gcp/lib.sh#L348-L353

Looks like mostly related to logging though https://github.com/kubernetes/k8s.io/blob/07a3a07626d8733550866abe799cf58241045a18/infra/gcp/lib.sh#L402-L412
